app/src/main/java/com/activities/SettingsActivity.java

```diff
@@ -0,0 +1,103 @@
+package com.activities;
+
+import com.R;
+import com.libs.globaldata.GlobalData;
+
+import android.content.SharedPreferences;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.CompoundButton;
+import android.widget.LinearLayout;
+import android.widget.Spinner;
+import android.widget.Switch;
+
+import androidx.appcompat.app.ActionBar;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.preference.PreferenceFragmentCompat;
+
+import java.io.File;
+
+public class SettingsActivity extends AppCompatActivity {
+
+    private LinearLayout layout;
+    private Spinner chooseModelSpinner;
+    private SharedPreferences userSettings;
+    private SharedPreferences.Editor editor;
+    private String[] models;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_settings);
+        layout = findViewById(R.id.settingsLayout);
+        chooseModelSpinner = findViewById(R.id.chooseModelSpinner);
```

Comment:
I think it should be called modelChoiceSpinner. chooseModelSpinner kind of suggests that we are choosing Spinner.